front/src/applications/editor/components/LayersModal.tsx

```diff
@@ -123,24 +122,34 @@ const LayersModal: FC<LayersModalProps> = ({
             const layerKey = layers.join('-');
             const count = sum(layers.map((id) => selectionCounts[id] || 0));
             const disabled = frozenLayers && layers.some((id) => frozenLayers.has(id));
+            const checked = layers.every((id) => selectedLayers.has(id));
             return (
               <div className="col-lg-6" key={`${layerKey}-${count}-${disabled}`}>
                 <div className="d-flex align-items-center mt-2">
                   <SwitchSNCF
                     type="switch"
-                    onChange={() =>
-                      setSelectedLayers((set) => {
-                        const newSet = new Set(set);
+                    onChange={() => {
+                      const newSelectedLayersList = () => {
```

Comment:
Here as you are using a function, you will re-compute the newSelectedLayersList 3 times. It would be better to compute it only once and then use it several times. You can also use a reduce to create it :)

```suggestion
const newSelectedLayersList = layers.reduce((result, layer) => {
    if (result.has(layer)) result.delete(layer);
    else result.add(layer);
    return result;
  }, new Set(selectedLayers));
```